### DISCUSSION TITLE
help creating hookfile

### DISCUSSION THREAD

**Discussion by kraweuschuasta** [adnanh/webhook#677] [Q&A] [answered]
Hi and good evening,

after creating and executing a very basic webhook 
```json
  {
    "id": "fronius1off",
    "execute-command": "/home/pi/Desktop/webhooks/fronius1off.sh",
    "command-working-directory": "/home/pi/Desktop/webhooks",
  },
```
I would like to create an extended version which exceeds my knowledge so far a bit...

I would like to add 2 parameters - a Number 1...4 and a String ON or OFF to the shellscript that gets executed.

So webhook http://192.168.1.111:9000/hooks/fronius?1&ON 
should start shellscript  fronius.sh 1 ON

Now I am stuck after hours and hours playing with pass-arguments-to-command and or clauses and "match" but I can't get it to work, I always get: "fronius got matched, but didn't get triggered because the trigger rules were not satisfied"

manyTHX for any help!
Wolfgang

**Comment by moorereason** [comment, marked as answer]
Use key=value pairs for the parameters: `http://192.168.1.111:9000/hooks/fronius?node=1&state=ON`

```json
  {
    "id": "fronius",
    "execute-command": "/home/pi/Desktop/webhooks/fronius.sh",
    "command-working-directory": "/home/pi/Desktop/webhooks",
    "pass-arguments-to-command": [
      {
        "source": "url",
        "name": "node"
      },
      {
        "source": "url",
        "name": "state"
      }
    ]
  }
```

**Reply by kraweuschuasta** [reply]
Super it works finally.
the key-value pairs AND using "url" as source did the trick!

THX a lot, 

Wolfgang
